### ISSUE TITLE
Changes on sub-models are prevented

### ISSUE THREAD

**Issue by joseandrespg** [sylvainpolletvillard/ObjectModel#43]
Hi, 

I think we found a bug. It happens when we try to modify a property in a sub-model, after the assignation, the new value doesn't get saved in the object and still printing the original value.

This example will reproduce the bug:

```javascript

var Model = require("objectmodel");

var Extra = new Model({
    status: String
});

var Order = new Model({
    sku: String,
    extra: Extra
});

var input = { sku: "ABC123", extra: { status: 'bbbb' }};

const o = new Order(input);
o.extra.status = 'aaaa';

console.log(o.extra.status); // prints 'bbbb' instead of 'aaaa'
```

Thanks for your time!

**Comment by sylvainpolletvillard** [comment]
Hi,

I reproduced the issue and will look into it tonight. It is related to automatic casting.

Easy workaround in the meantime:
```
var input = { sku: "ABC123", extra: Extra({ status: 'bbbb' })};
```

**Comment by sylvainpolletvillard** [comment]
Fixed in v2.6.1, thanks for the report

**Comment by joseandrespg** [comment]
Hi, I can confirm that the fix worked for us.

Cheers! :+1:
